### PR TITLE
docs: add React examples for charts

### PR DESCRIPTION
**Pull request by tomivirkki**
Extracted from https://github.com/vaadin/docs/pull/2632

### CODE REVIEW

File: frontend/demo/component/charts/react/charts-overview.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      <Chart theme={theme} style={chartStyle} type="pie" tooltip>
+        <ChartSeries title="Brands" values={[38, 24, 15, 8]} />
+      </Chart>
+
+      <Chart theme={theme} style={chartStyle} polar>
+        <ChartSeries
+          type="column"
+          title="Column"
+          values={[8, 7, 6, 5, 4, 3, 2, 1]}
+          // additionalOptions={{ pointPlacement: 'between' }}
```
Comment (web-padawan) [inline frontend/demo/component/charts/react/charts-overview.tsx:76]: Should this be removed?

**Review by web-padawan — APPROVED** [review]

File: frontend/demo/component/charts/react/charts-overview.tsx (hunk 1 above)
Comment (tomivirkki) [reply frontend/demo/component/charts/react/charts-overview.tsx]: Ah, looks like the `additionalOptions` were missing from the other charts also. Added all the missing options.

File: frontend/demo/component/charts/react/charts-overview.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          title="Poud sterling"
+          values={[32, 21, 8, 12, 32, 21, 12, 30, 25, 19, 26, 15]}
+        />
+      </Chart>
+
+      <Chart theme={theme} style={chartStyle} type="pie" tooltip additionalOptions={pieOptions}>
+        <ChartSeries title="Brands" values={pieValues} />
+      </Chart>
+
+      <Chart theme={theme} style={chartStyle} polar>
```
Comment (vursen) [inline frontend/demo/component/charts/react/charts-overview.tsx:124]: This chart seems to be missing `additionalOptions` which are present in the TypeScript example.

File: frontend/demo/component/charts/react/charts-overview.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+const pieValues: PointOptionsObject[] = [
+  { name: 'Chrome', y: 38 },
+  { name: 'Firefox', y: 24 },
+  { name: 'Edge', y: 15, sliced: true, selected: true },
+  { name: 'Internet Explorer', y: 8 },
+];
+
+const seriesOptions: SeriesOptionsType = {
+  pointPlacement: 'between',
+  type: 'column',
```
[the remark this answers is not in the document]
Comment (tomivirkki) [reply frontend/demo/component/charts/react/charts-overview.tsx:72]: It seems to be required by `SeriesOptionsType `

File: frontend/demo/component/charts/react/charts-overview.tsx (hunk 2 above)
Comment (tomivirkki) [reply frontend/demo/component/charts/react/charts-overview.tsx]: Added

**Pull request merged**